examples/bert/run_pretraining.py

```diff
@@ -398,16 +420,38 @@ def main(_):
         num_warmup_steps=FLAGS.num_warmup_steps,
         num_train_steps=FLAGS.num_train_steps,
     )
+    optimizer = keras.optimizers.Adam(learning_rate=learning_rate_schedule)
+
+    if FLAGS.skip_restore:
+        # Error out if users want to skip restoring but checkpoints are found
+        # in the directory.
+        if len(os.listdir(FLAGS.checkpoint_save_path)) > 0:
+            raise RuntimeError(
+                "Please point `checkpoint_save_path` to an "
+                "empty directory if you want to skip restoring model from "
+                "saved checkpoints. Checkpoints are found under "
+                f"{FLAGS.checkpoint_save_path} "
+            )
+
+    callbacks = []
+    if FLAGS.checkpoint_save_path is not None:
+        checkpoint_path = FLAGS.checkpoint_save_path + "/checkpoint_{epoch:2d}"
```

Comment:
Have you tested this out? Just looking through the backup and restore code this dir, it looks like this directory get's joined with other directory names and passed directly to `tf.train.CheckpointManager`, which I don't think supports this `{epoch:2d}` format you are using?

Make sure you actually `ls` the directory you are saving too when testing this out.

Reply:
Yea, this was a legacy code. this style works well with the model checkpoint callback, but not the backupandrestore. I will update the code.